go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -982,74 +980,16 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 	// Note(adrianc): DPU deployments are expected to support the new shared gateway changes, upgrade flow
 	// is not needed. Future upgrade flows will need to take DPUs into account.
 	if config.OvnKubeNode.Mode != types.NodeModeDPUHost {
-		// Upgrade for Node. If we upgrade workers before masters, then we need to keep service routing via
-		// mgmt port until masters have been updated and modified OVN config. Run a goroutine to handle this case
-		upgradeController := upgrade.NewController(nc.client, nc.watchFactory)
-		initialTopoVersion, err := upgradeController.GetTopologyVersion(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to get initial topology version: %w", err)
-		}
-		klog.Infof("Current control-plane topology version is %d", initialTopoVersion)
-
 		bridgeName := ""
 		if config.OvnKubeNode.Mode == types.NodeModeFull {
 			bridgeName = nc.gateway.GetGatewayBridgeIface()
-
-			needLegacySvcRoute := true
-			if (initialTopoVersion >= types.OvnHostToSvcOFTopoVersion && config.GatewayModeShared == config.Gateway.Mode) ||
-				(initialTopoVersion >= types.OvnRoutingViaHostTopoVersion) {
-				// Configure route for svc towards shared gw bridge
-				// Have to have the route to bridge for multi-NIC mode, where the default gateway may go to a non-OVS interface
-				if err := configureSvcRouteViaBridge(nc.routeManager, bridgeName); err != nil {
-					return err
-				}
-				needLegacySvcRoute = false
-			}
-
-			// Determine if we need to run upgrade checks
-			if initialTopoVersion != types.OvnCurrentTopologyVersion {
-				if needLegacySvcRoute {
-					klog.Info("System may be upgrading, falling back to legacy K8S Service via management port")
-					// add back legacy route for service via management port
-					link, err := util.LinkSetUp(types.K8sMgmtIntfName)
-					if err != nil {
-						return fmt.Errorf("unable to get link for %s, error: %v", types.K8sMgmtIntfName, err)
-					}
-					var gwIP net.IP
-					for _, subnet := range config.Kubernetes.ServiceCIDRs {
-						if utilnet.IsIPv4CIDR(subnet) {
-							gwIP = mgmtPortConfig.ipv4.gwIP
-						} else {
-							gwIP = mgmtPortConfig.ipv6.gwIP
-						}
-						subnet := *subnet
-						nc.routeManager.Add(netlink.Route{
-							LinkIndex: link.Attrs().Index,
-							Gw:        gwIP,
-							Dst:       &subnet,
-							MTU:       config.Default.RoutableMTU,
-						})
-					}
-				}
+			// Configure route for svc towards shared gw bridge
+			// Have to have the route to bridge for multi-NIC mode, where the default gateway may go to a non-OVS interface
+			if err := configureSvcRouteViaBridge(nc.routeManager, bridgeName); err != nil {
```

Comment:
remind me why we need to do this always.. is it one of those differences between sgw and lgw? seems like it used to be a sgw mode only thing..

Reply:
It wasn't a sgw mode only, this was the condition:
```
		if (initialTopoVersion >= types.OvnHostToSvcOFTopoVersion && config.GatewayModeShared == config.Gateway.Mode) ||
				(initialTopoVersion >= types.OvnRoutingViaHostTopoVersion) {
				// Configure route for svc towards shared gw bridge
				// Have to have the route to bridge for multi-NIC mode, where the default gateway may go to a non-OVS interface
				if err := configureSvcRouteViaBridge(nc.routeManager, bridgeName); err != nil {
					return err
				}
			}
```
which could be simplified to:
```
if (initialTopoVersion >= types.OvnRoutingViaHostTopoVersion) {...
```
which means it always happen. The behavior changed from being sgw only here: https://github.com/openshift/ose-ovn-kubernetes/commit/db1ec12165d5f7bde0008c7bca185670514b5b9e

Reply:
ah thanks pat you are right